rust/xaynet-server/src/state_machine/phases/idle.rs

```diff
@@ -39,38 +41,50 @@ where
 {
     const NAME: PhaseName = PhaseName::Idle;
 
-    async fn run(&mut self) -> Result<(), PhaseStateError> {
-        info!("updating the keys");
-        self.gen_round_keypair();
-
-        info!("updating round thresholds");
-        self.update_round_thresholds();
-
-        info!("updating round seeds");
-        self.update_round_seed();
+    impl_phase_method_for_phasestate! {
+        async fn process(self_: &mut PhaseState<Idle, S>) -> Result<(), PhaseStateError> {
+            info!("updating the keys");
+            self_.gen_round_keypair();
+
+            info!("updating round probabilities");
+            self_.update_round_probabilities();
+
+            info!("updating round seed");
+            self_.update_round_seed();
+
```

Comment:
i moved the invalidation broadcasts to the error phase, which is the earliest possible time. to avoid store waiting times in the error phase itself, i also moved the store readiness to the `next()` method (it somewhat belongs there anyways, because waiting for the store is a necessary condition to transition to the next phase. it is also a passive task, there is nothing the state machine can do to make the store ready, so it doesn't really fit into the `process()` of the error phase as well).